packages/react/src/components/Modal/index.tsx

```diff
@@ -69,91 +72,41 @@ export const Modal = (props: ModalProps) => {
 };
 
 export const ModalInner = (props: ModalProps) => {
-  const ref = React.useRef<HTMLDialogElement>(null);
-
-  useEffect(() => {
-    if (props.isOpen) {
-      ref.current?.showModal();
-    } else {
-      ref.current?.close();
-    }
-  }, [props.isOpen]);
-
-  useEffect(() => {
-    ref.current?.addEventListener("close", () => props.onClose(), {
-      once: true,
-    });
-
-    const onClick = (event: MouseEvent) => {
-      const rect = ref.current!.getBoundingClientRect();
-      const isInDialog =
-        rect.top <= event.clientY &&
-        event.clientY <= rect.top + rect.height &&
-        rect.left <= event.clientX &&
-        event.clientX <= rect.left + rect.width;
-      if (!isInDialog) {
-        props.onClose();
-      }
-    };
-
-    if (props.closeOnClickOutside) {
-      ref.current?.addEventListener("click", onClick);
-    }
-
-    const preventClose = (event: Event) => {
-      event.preventDefault();
-    };
-
-    if (props.preventClose) {
-      ref.current?.addEventListener("cancel", preventClose);
-    }
-
-    return () => {
-      ref.current?.removeEventListener("click", onClick);
-      ref.current?.removeEventListener("click", preventClose);
-    };
-  }, [props.isOpen]);
-
-  if (!props.isOpen) {
-    return null;
-  }
-
   return (
-    <>
-      {createPortal(
-        <>
-          <div aria-hidden={true} className="c__modal__backdrop" />
-          <dialog
-            ref={ref}
-            className={classNames("c__modal", "c__modal--" + props.size)}
-          >
-            {!props.hideCloseButton && !props.preventClose && (
-              <div className="c__modal__close">
-                <Button
-                  icon={<span className="material-icons">close</span>}
-                  color="tertiary-text"
-                  size="small"
-                  onClick={() => props.onClose()}
-                />
-              </div>
-            )}
-            {props.titleIcon && (
-              <div className="c__modal__title-icon">{props.titleIcon}</div>
-            )}
-            {props.title && (
-              <div className="c__modal__title">{props.title}</div>
-            )}
-
-            {/* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex */}
-            <div className="c__modal__content" tabIndex={0}>
-              {props.children}
-            </div>
-            <ModalFooter {...props} />
-          </dialog>
-        </>,
-        document.getElementById("c__modals-portal")!,
+    <ReactModal
+      isOpen={props.isOpen}
+      onRequestClose={() => {
+        if (!props.preventClose) {
+          props.onClose();
+        }
+      }}
+      parentSelector={() => document.getElementById("c__modals-portal")!}
+      overlayClassName="c__modal__backdrop"
+      className={classNames(MODAL_CLASS, `${MODAL_CLASS}--${props.size}`)}
+      shouldCloseOnOverlayClick={!!props.closeOnClickOutside}
```

Comment:
There is also a prop to prevent to close the modal by pressing the Escape key.

I think it could be useful as in Richie, we have a use case where we want to display a close button (so here we cannot set `preventClose` prop to `true`) but we want to prevent the user to misclick outside of the modal or mispress the escape key.

Reply:
Done :)